src/main/java/org/myrobotlab/swing/ProgramABGui.java

```diff
@@ -158,23 +188,74 @@ public void actionPerformed(ActionEvent event) {
 
   @Override
   public void subscribeGui() {
+    subscribe("publishRequest");
+    subscribe("publishText");
+    subscribe("publishOOBText");
   }
 
   @Override
   public void unsubscribeGui() {
+    unsubscribe("publishText");
+    unsubscribe("publishRequest");
+    unsubscribe("publishOOBText");
+  }
+
+  public void onText(String text) {
+    SwingUtilities.invokeLater(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          responseKit.insertHTML(responseDoc, responseDoc.getLength(), "<i><b><br>" + botName.getText() + "</b>: " + text.replaceAll("\\<.*?\\>", " ").trim(), 0, 0, Tag.I);

```

Comment:
it is hidden :) the last parameter of insertHTML() is used to close tags, useful to save some code line. Tag.I , Tag.[...] . I just discovered that.

Reply:
Are you sure?
I thought it would be both opening&closing tags in addition to Tag.I ?
Maybe I can't remeber how this is used from the last time I used it. Will test later if required.

Reply:
yup in this syntax the last parameter will add a closed I tag